### PR TITLE
 Adding Hadoop DSL support for sending multiple topics with KafkaPushJob

### PR DESCRIPTION
**Pull request by tzuhanjan**
Tested via snapshot builds.

### CODE REVIEW

File: hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/KafkaPushJob.groovy [hunk 1]
```diff
@@ -179,4 +185,26 @@ class KafkaPushJob extends HadoopJavaJob {
...
+
+  /**
+   * DSL usesDisableAuditing method causes multi.topic=value to be set in the job file.
+   *
+   * @param multiTopic
+   */
+  @HadoopDslMethod
+  void usesMultiTopic(Boolean multiTopic) {
+    this.multiTopic = multiTopic;
+    setJobProperty("multi.topic", multiTopic);
```
Comment (pranayhasan) [inline hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/KafkaPushJob.groovy]: If the property is not set, don't you want to set the default values on the DSL side? If not, can you confirm if the null checks are happening on the Azkaban Job side?

Comment (tzuhanjan) [reply hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/KafkaPushJob.groovy]: There is a default value set in the Azkaban job itself to set this to "False" such that it's backwards compatible. Is the convention to set a default value? The other properties added for this specific DSL haven't added any defaults in the DSL, only in the job.

File: hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/KafkaPushJob.groovy [hunk 2]
```diff
@@ -179,4 +185,26 @@ class KafkaPushJob extends HadoopJavaJob {
     this.disableAuditing = disableAuditing;
     setJobProperty("disable.auditing", disableAuditing);
   }
+
+  /**
+   * DSL usesDisableAuditing method causes multi.topic=value to be set in the job file.
```
Comment (tzuhanjan) [inline hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/KafkaPushJob.groovy]: c+p error: usesDisableAuditing --> usesMultiTopic

File: hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/KafkaPushJob.groovy [hunk 3]
```diff
@@ -179,4 +185,26 @@ class KafkaPushJob extends HadoopJavaJob {
...
+   * @param multiTopic
+   */
+  @HadoopDslMethod
+  void usesMultiTopic(Boolean multiTopic) {
+    this.multiTopic = multiTopic;
+    setJobProperty("multi.topic", multiTopic);
+  }
+
+  /**
+   * DSL usesDisableAuditing method causes whitelisted.topics=value to be set in the job file.
```
Comment (tzuhanjan) [inline hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/KafkaPushJob.groovy]: c+p error: usesDisableAuditing --> usesMultiTopic

Comment (pranayhasan) [reply hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/KafkaPushJob.groovy]: usesDisableAuditing --> usesWhitelistedTopics

File: hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/KafkaPushJob.groovy (hunk 1 above)
Comment (pranayhasan) [reply hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/KafkaPushJob.groovy]: If you can update the PR's javadocs accordingly, I'm good.

**Comment by pranayhasan** [comment]
Hey @tzuhanjan the PR looks good. If you can address fixing the javadocs, I'm gonna merge this.
Thanks

**Pull request merged**